db/schema_applier.cc

```diff
@@ -0,0 +1,923 @@
+                          const std::unordered_map<sstring, table_selector>& tables_per_keyspace)
+{
+    std::map<table_id, schema_mutations> result;
+    for (auto&& [keyspace_name, sel] : tables_per_keyspace) {
+        if (!sel.tables.contains(kind)) {
+            continue;
+        }
+        for (auto&& table_name : sel.tables.find(kind)->second) {
+            auto qn = qualified_name(keyspace_name, table_name);
+            auto muts = co_await read_table_mutations(proxy, qn, get_table_holder(kind));
+            auto id = table_id_from_mutations(muts);
+            if (id) {
+                result.emplace(std::move(*id), std::move(muts));
+            }
+        }
+    }
+    co_return result;
+}
+
+// Extracts the names of tables affected by a schema mutation.
+// The mutation must target one of the tables in schema_tables_holding_schema_mutations().
+static
+table_selector get_affected_tables(const sstring& keyspace_name, const mutation& m) {
+    const schema& s = *m.schema();
+    auto get_table_name = [&] (const clustering_key& ck) {
+        // The first component of the clustering key in each table listed in
+        // schema_tables_holding_schema_mutations contains the table name.
+        return value_cast<sstring>(utf8_type->deserialize(ck.get_component(s, 0)));
+    };
+    table_selector result;
+    if (m.partition().partition_tombstone()) {
+        slogger.trace("Mutation of {}.{} for keyspace {} contains a partition tombstone",
+                      m.schema()->ks_name(), m.schema()->cf_name(), keyspace_name);
+        result.all_in_keyspace = true;
+    }
+    for (auto&& e : m.partition().row_tombstones()) {
+        const range_tombstone& rt = e.tombstone();
+        if (rt.start.size(s) == 0 || rt.end.size(s) == 0) {
+            slogger.trace("Mutation of {}.{} for keyspace {} contains a multi-table range tombstone",
+                          m.schema()->ks_name(), m.schema()->cf_name(), keyspace_name);
+            result.all_in_keyspace = true;
+            break;
+        }
+        auto table_name = get_table_name(rt.start);
+        if (table_name != get_table_name(rt.end)) {
+            slogger.trace("Mutation of {}.{} for keyspace {} contains a multi-table range tombstone",
+                          m.schema()->ks_name(), m.schema()->cf_name(), keyspace_name);
+            result.all_in_keyspace = true;
+            break;
+        }
+        result.add(table_name);
+    }
+    for (auto&& row : m.partition().clustered_rows()) {
+        result.add(get_table_name(row.key()));
+    }
+    slogger.trace("Mutation of {}.{} for keyspace {} affects tables: {}, all_in_keyspace: {}",
+                  m.schema()->ks_name(), m.schema()->cf_name(), keyspace_name, result.tables, result.all_in_keyspace);
+    return result;
+}
+
+future<schema_result>
+static read_schema_for_keyspaces(distributed<service::storage_proxy>& proxy, const sstring& schema_table_name, const std::set<sstring>& keyspace_names)
```

Comment:
The split seems a bit arbitrary to me. For example, why is read_schema_for_keyspaces() in schema_applier.cc, and other schema reading methods still in schema_tables.cc?

Reply:
read_schema_for_keyspaces is a static function needed only in this file. what other functions you have in mind? read_schema_partition_for_keyspace for instance is used in multiple places and is in header file. There is no need to expose functions if they are not used. 

The split is based on what code is needed only for merging and it's put into a single file.